src/System.Memory/ref/System.Memory.cs

```diff
@@ -87,6 +87,7 @@ public static class SpanExtensions
     {
          public static int IndexOf<T>(this Span<T> span, T value) where T:struct, IEquatable<T> { throw null; }
```

Comment:
Yes, it should be pretty equivalent - once the JIT team implements the planned struct enregistration that handles Span well.